apps/game_backend/lib/game_backend/campaigns.ex

```diff
@@ -13,9 +13,9 @@ defmodule GameBackend.Campaigns do
   def get_campaigns() do
     campaigns =
       Repo.all(from(c in Campaign))
-      |> Repo.preload(levels: [units: :items])
+      |> Repo.preload(levels: [:currency_rewards, units: [:items, :character]])
 
-    if Enum.empty?(campaigns), do: {:error, :no_campaigns}, else: campaigns
+    if Enum.empty?(campaigns), do: {:error, :no_campaigns}, else: Enum.sort_by(campaigns, & &1.campaign_number)
```

Comment:
why not creating a single query that does all instead of pipe the results?
```
q = 
   from c in Campaign,
   preload: [levels: [:currency_rewards, units: [:items, :character]]],
   order_by: [asc: c.campaign_number]
```
do you think the pipes are more readable?

Reply:
No, not at all :P your solution is much better.